src/main/java/EdgeSaveParser.java

```diff
@@ -0,0 +1,116 @@
+import java.io.*;
+import java.util.*;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.swing.*;
+
+
+public class EdgeSaveParser extends EdgeConvertFileParser {
+ 
+   private BufferedReader br;
+    private int numFigure;
+    public static Logger log = LogManager.getLogger(EdgeSaveParser.class);
+
+    public EdgeSaveParser(File parseFile){
+        super();
+        numFigure = 0;
+        this.openFile(parseFile);
+
+   }
+
+    public void parseSaveFile() throws IOException { //this method is unclear and confusing in places
+        log.debug("Changing save file and parsing it");
+        StringTokenizer stTables, stNatFields, stRelFields, stField;
+        EdgeTable tempTable;
+        EdgeField tempField;
+        String currentLine = br.readLine();
+        currentLine = br.readLine(); //this should be "Table: "
+        while (currentLine.startsWith("Table: ")) {
+           numFigure = Integer.parseInt(currentLine.substring(currentLine.indexOf(" ") + 1)); //get the Table number
+           currentLine = br.readLine(); //this should be "{"
+           currentLine = br.readLine(); //this should be "TableName"
+            String tableName = currentLine.substring(currentLine.indexOf(" ") + 1);
+           tempTable = new EdgeTable(numFigure + DELIM + tableName);
+           
+           currentLine = br.readLine(); //this should be the NativeFields list
+           stNatFields = new StringTokenizer(currentLine.substring(currentLine.indexOf(" ") + 1), DELIM);
+            int numFields = stNatFields.countTokens();
+           for (int i = 0; i < numFields; i++) {
+              tempTable.addNativeField(Integer.parseInt(stNatFields.nextToken()));
+           }
+           
+           currentLine = br.readLine(); //this should be the RelatedTables list
+           stTables = new StringTokenizer(currentLine.substring(currentLine.indexOf(" ") + 1), DELIM);
+            int numTables = stTables.countTokens();
+           for (int i = 0; i < numTables; i++) {
+              tempTable.addRelatedTable(Integer.parseInt(stTables.nextToken()));
+           }
+           tempTable.makeArrays();
+           
+           currentLine = br.readLine(); //this should be the RelatedFields list
+           stRelFields = new StringTokenizer(currentLine.substring(currentLine.indexOf(" ") + 1), DELIM);
+           numFields = stRelFields.countTokens();
+  
+           for (int i = 0; i < numFields; i++) {
+              tempTable.setRelatedField(i, Integer.parseInt(stRelFields.nextToken()));
+           }
```

Comment:
3 for loops in a while loop. Wonder if we could use a map. Remember: nested loops are exponential!